src/jit/codegenxarch.cpp

```diff
@@ -3756,7 +3759,10 @@ void CodeGen::genCodeForCmpXchg(GenTreeCmpXchg* tree)
 // generate code for BoundsCheck nodes
 void CodeGen::genRangeCheck(GenTree* oper)
 {
-#ifdef FEATURE_SIMD
+#if defined(FEATURE_HW_INTRINSICS) && defined(FEATURE_SIMD)
```

Comment:
Shouldn't the standalone `GT_HWIntrinsic_CHK` case also be handled (at the very least, based on the `ifdefs` elsewhere)?

Reply:
So far, all the imm-intrinsics are SIMD. Do you think it is necessary?

Reply:
I think that all `#ifdef`s concerning the HWIntrinsic Check should be consistent. So they either all need to be under `#if defined(FEATURE_HW_INTRINSICS) && defined(FEATURE_SIMD)` or they should all be under `#ifdef FEATURE_HW_INTRINSICS`. Having a mix of both (when there isn't two actual distinctions of where the code applies) just makes it confusing (and prone to breakage) later down the road.

Reply:
Got it, I will add a standalone `#ifdef GT_HWIntrinsic_CHK`.